R/utils.R

```diff
@@ -171,6 +172,122 @@ list_init_empty <- function(x,
   list(x = x, sizes = sizes, null = empty_null, typed = empty_typed)
 }
 
+#' Replace `NULL` list elements
+#'
+#' @param x A list, but not a list-of.
+#'
+#' @param sizes An integer vector of sizes of the `x` elements.
+#'
+#' @param ptype One of:
+#'
+#'   - `NULL` to fill `NULL` elements with `unspecified()`.
+#'
+#'   - A `ptype` value to fill `NULL` elements with.
+#'
+#' @param size The size of the replacement value to fill with. Commonly either
+#'   `0L` or `1L`.
+#'
+#' @returns
+#' A named list holding `x` with `NULL`s replaced, and `sizes` updated to
+#' have the correct size where `NULL` elements were replaced.
+#'
+#' @noRd
+list_init_null <- function(x, sizes, ..., ptype = NULL, size = 1L) {
```

Comment:
The name made me thing think would initialise a list of `NULL`s. Maybe `list_replace_null()`?